### DISCUSSION TITLE
Is there a way to escape `::` in a test name?

### DISCUSSION THREAD

**Discussion by giovanni-guidini** [pytest-dev/pytest#11595] [Q&A]
We are starting rely quite heavily in asking pytest to run specific tests.
In general this is fine if you just escape the test name with singlequotes, for example the command below would work as expected.

```
pytest \
'tests/unit/components/test_components.py::test_from_dict' \
'tests/unit/test_report_file.py::test_get_item_exception[get_val0-expecting type int got <class '"'"'list'"'"'>]' \
'tests/unit/encryption/test_yaml_secret.py::test_yaml_secret_cases[wnb1smVbBvjHNFgEgK2RV9rfVhGygj3IAPOkWp3rO8o=-Im a banana]' \
```

However some tests have `::` in the parametrized string, and I believe that is throwing pytest off
Example:
```
$ pytest 'tests/unit/encryption/test_yaml_secret.py::test_yaml_secret_cases[v2::FpVtGkoVY7ACYicwfD1QP3oAPOmJj+eM24l2IkWfiNMtVInE8JVNJnJZJqIKmVLy7LW479oJuIfFRWI2COIJrA==-gitlab/1234/876/Talkthetalkwalkthewalk]' 

ERROR: not found: ROOT_DIR/tests/unit/encryption/test_yaml_secret.py::test_yaml_secret_cases[v2::FpVtGkoVY7ACYicwfD1QP3oAPOmJj+eM24l2IkWfiNMtVInE8JVNJnJZJqIKmVLy7LW479oJuIfFRWI2COIJrA==-gitlab/1234/876/Talkthetalkwalkthewalk]
(no name 'ROOT_DIR/tests/unit/encryption/test_yaml_secret.py::test_yaml_secret_cases[v2::FpVtGkoVY7ACYicwfD1QP3oAPOmJj+eM24l2IkWfiNMtVInE8JVNJnJZJqIKmVLy7LW479oJuIfFRWI2COIJrA==-gitlab/1234/876/Talkthetalkwalkthewalk]' in any of [<Module test_yaml_secret.py>])
```

Is there a way to escape this sequence in the parametrized sequence?

**Comment by audiodude** [comment]
So the question is, can `::` appear in the parameterization specification of a node ID. I looked through https://docs.pytest.org/en/7.1.x/example/markers.html?highlight=single%20test#selecting-tests-based-on-their-node-id and some of the code and I didn't see anything about escaping.
